dali/imgcodec/util/convert.cc

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cassert>
+#include <utility>
+#include "dali/core/static_switch.h"
+#include "dali/imgcodec/util/convert.h"
+#include "dali/kernels/slice/slice_cpu.h"
+#include "dali/kernels/common/utils.h"
+
+namespace dali {
+namespace imgcodec {
+
+template <int static_ndim = -1, typename Out, typename In, typename ConvertFunc>
+void Convert(Out *out, const int64_t *out_strides,
+             const In *in, const int64_t *in_strides,
+             const int64_t *size, int ndim,
+             ConvertFunc &&func) {
+  if constexpr (static_ndim < 0) {
+    VALUE_SWITCH(ndim, NDim, (0, 1, 2, 3, 4),
+      (Convert<NDim>(out, out_strides, in, in_strides, size, NDim,
+                     std::forward<ConvertFunc>(func));
+      return;), ()
+    );  // NOLINT
+  }
+
+  int64_t extent = size[0];
+  int64_t in_stride = in_strides[0];
+  int64_t out_stride = out_strides[0];
+
+  if constexpr (static_ndim == 0) {
```

Comment:
Do we ever get here if we have the case for static_ndim == 1 which happens earlier?
I'm not sure if I fully understand.

Reply:
Theoretically, we may start with a scalar.